### ISSUE TITLE
Mixed activation function fails when device='cuda'

### ISSUE THREAD

**Issue by tsrobinson** [tsrobinson/SyGNet#4]
When training on GPU:

`model_gpu = SygnetModel(mode = "wgan")`
`model.fit(data = sim_data, epochs = 10, device='cuda')`

RuntimeError:
Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument index in method wrapper__index_select)

...

File ~/anaconda3/envs/dl/lib/python3.9/site-packages/sygnet/sygnet_models.py:39, in _MixedActivation.forward(self, input)
     [37](file:///home/tsr/anaconda3/envs/dl/lib/python3.9/site-packages/sygnet/sygnet_models.py?line=36)         mixed_out.append(torch.sigmoid(torch.index_select(input, 1, index_.type(torch.int32))))
     [38](file:///home/tsr/anaconda3/envs/dl/lib/python3.9/site-packages/sygnet/sygnet_models.py?line=37)     elif self.funcs[number] == 'identity':
---> [39](file:///home/tsr/anaconda3/envs/dl/lib/python3.9/site-packages/sygnet/sygnet_models.py?line=38)         mixed_out.append(self.identity(torch.index_select(input, 1, index_.type(torch.int32))))
     [40](file:///home/tsr/anaconda3/envs/dl/lib/python3.9/site-packages/sygnet/sygnet_models.py?line=39) return torch.cat(mixed_out,1)

**Comment by tsrobinson** [comment]
Fails even when `mixed_activation=False`
